### PR TITLE
Integrated news page into assistant-canvas directory

### PR DESCRIPTION
**Pull request by jack-y-xu**

### CODE REVIEW

**Comment by yukyhaha** [comment]
Can you add a description about what most file changes are about?

File: news/public/js/assistant.js [hunk 1]
```diff
@@ -50,28 +50,89 @@ class Assistant {
   }
 }
 
+async function getSentimentsAndSort(articles) {
+  //add articles to a json obj
```
Comment (asam01) [inline news/public/js/assistant.js]: nit: add space before comment
// add articles to a json obj

File: news/public/js/assistant.js [hunk 2]
```diff
@@ -50,28 +50,89 @@ class Assistant {
...
+  var jsonObject = {};
+  jsonObject.documents = [];
+  for(i=0;i<articles.length;i++) {
```
Comment (asam01) [inline news/public/js/assistant.js]: nit: add spaces
for(i = 0; i < artic....; i++)

File: news/public/js/assistant.js [hunk 3]
```diff
@@ -50,28 +50,89 @@ class Assistant {
...
+    var docObject = {};
+    docObject.id = i+1;
+    docObject.language = "en";
+    docObject.text = article.name + ". " + article.description;
+    jsonObject.documents.push(docObject);
+  }
+  //make obj into json
+  var docJSON = JSON.stringify(jsonObject);
+  console.log(articles);
+  console.log(docJSON);
```
Comment (asam01) [inline news/public/js/assistant.js]: did you mean to leave these in? are these for debugging?

File: news/public/js/assistant.js [hunk 4]
```diff
@@ -50,28 +50,89 @@ class Assistant {
...
+  //get the sentiment values
+  var documents = {};
+  fetch("https://microsoft-text-analytics1.p.rapidapi.com/sentiment", {
+    "method": "POST",
+    "headers": {
+      "x-rapidapi-host": "microsoft-text-analytics1.p.rapidapi.com",
+      "x-rapidapi-key": "968db1c3aamsh38e8276ca5be196p1bab0ejsn0973e5df894d",
```
Comment (asam01) [inline news/public/js/assistant.js]: not sure if we want this to be private - maybe add it to a separate file and put that file in gitignore. if it doesn't matter if the API key is public, then ignore this

File: news/public/js/assistant.js [hunk 5]
```diff
@@ -50,28 +50,89 @@ class Assistant {
...
+  }).then(response => response.json()).then(response => {
+    documents = response.documents;
+  });
+  //now assign the scores to the actual articles
+  for(i=0;i<articles.length;i++) {
+    var doc = documents[i];
+    articles[i].positivity = doc.documentScores.positive - doc.documentScores.negative;
+  }
+  //now sort the articles and return them
+  console.log(typeof(articles));
```
Comment (asam01) [inline news/public/js/assistant.js]: debugging?

File: news/public/js/assistant.js [hunk 6]
```diff
@@ -50,28 +50,89 @@ class Assistant {
...
+  articles = [].slice.call(articles).sort(function(a,b){
+    if (a.positivity == b.positivity) return 0;
```
Comment (asam01) [inline news/public/js/assistant.js]: consider using `===`. check out this article: https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons

**Pull request closed without merging**